### PR TITLE
`Development`: Extend ngx-charts documentation section

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

This PR extends the documentation section regarding ngx-charts with some additional information.

### Steps for Testing

1. Check the documentation under [Coding and design guidelines/Client/14. Chart Instantiation](https://artemis-platform--5341.org.readthedocs.build/en/5341/dev/guidelines/client/#chart-instantiation)

#### Code Review
- [x] Review 1
- [x] Review 2

### Screenshots

image
image

### CODE REVIEW

File: docs/dev/guidelines/client.rst [hunk 1]
```diff
@@ -282,22 +282,69 @@ Here are a few tips when using this framework:
...
+
+          [dataLabelFormatting]="formatDataLabel"
+
+       with
+
+       .. code-block:: ts
+
+          formatDataLabel(averageScore: number): string {
+                  return averageScore + '%';
+              }
```
Comment (b-fein) [inline docs/dev/guidelines/client.rst:298]:
```suggestion
          formatDataLabel(averageScore: number): string {
              return averageScore + '%';
          }
```
indentation

File: docs/dev/guidelines/client.rst [hunk 2]
```diff
@@ -282,22 +282,69 @@ Here are a few tips when using this framework:
...
+
+       appends a percentage sign to the data label.
+
+       .. TIP::
+           The method is passed to the framework itself and executed there. This means on runtime, it does not have access to global variables of the component it is originally implemented in.
```
Comment (b-fein) [inline docs/dev/guidelines/client.rst:303]:
```suggestion
           The method is passed to the framework itself and executed there. This means that at runtime it does not have access to global variables of the component it is originally implemented in.
```

File: docs/dev/guidelines/client.rst [hunk 3]
```diff
@@ -282,22 +282,69 @@ Here are a few tips when using this framework:
...
+         So, these names have to be translated every time the user switches the language settings.
+         In this case, inject the ``TranslateService`` to the underlying component and subscribe to the ``onLangChange`` event emitter:
+
+       .. code-block:: ts
+
+           constructor(private translateService: TranslateService) {
+                this.translateService.onLangChange.subscribe(() => {
+                    this.updateXAxisLabel(); // a method re-assigning the names of the objects to the translated string
+                });
+           }
```
Comment (b-fein) [inline docs/dev/guidelines/client.rst:346]:
```suggestion
           constructor(private translateService: TranslateService) {
               this.translateService.onLangChange.subscribe(() => {
                   this.updateXAxisLabel(); // a method re-assigning the names of the objects to the translated string
               });
           }
```
indentation

**Review by b-fein — COMMENTED** [review]
Only minimal formatting suggestions, looks good otherwise.

**Review by b-fein — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
The documentation is clear

**Review by Dominik-Weinzierl — APPROVED** [review]
All chart details i used during my development a mentioned in this documentation - good job :+1:

**Review by ge65cer — APPROVED** [review]
Read through the documentation and approve. Big thanks for adding the `::ng-deep` warning.

**Review by Marvin-Vitcu — APPROVED** [review]
Checked the documentation. Very understandable, with no spelling or grammar mistakes.

**Pull request merged**